pkg/networkmanager/firewall.jsx

```diff
@@ -270,7 +272,12 @@ export class Firewall extends React.Component {
             );
         }
 
-        var addServiceAction = <button className="btn btn-primary pull-right" onClick={this.onAddServices}>{_("Add Services…")}</button>;
+        var addServiceAction;
+        var classes = "btn btn-primary pull-right";
+        if (this.state.firewall.readonly)
+            addServiceAction = <button className={classes} disabled title={_("You are not authorized to modify the firewall.")}>{_("Add Services…")}</button>;
```

Comment:
This is using the standard browser tooltips (white on black) which are "wrong". I suggest using `update_privileged()` for that, or if that somehow is impractical, use `.tooltip('fixTitle')` (but that requires jQuery) or our `Tooltip` react component.

Reply:
`update_privileged()` also depends on jQuery, which I don't want to pull into `firewall.jsx`. I've used the React component, which unfortunately needed some minor tweaking. I've added the two commits on here as well.